### PR TITLE
Consume Microsoft.DotNet.XUnitConsoleRunner for netcoreapp

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes "I would prefer we get CoreFX to switch to the xunit runner we need rather than figuring out how to replace it in flight (between corefx and helix), but if we have to replace it, fine. Maybe Viktor can help?"

I advise you to not read further. Sometimes not knowing things is just better.

Our funny test runner matrix:

| TFM | Runner |
| ----- | -------- |
| netcoreapp | XUnitConsoleRunner |
| netfx | xunit.runner.console |
| uap | XUnitRunnerUap (XUnitConsoleRunner cross compiled) |
| *aot | ILC version of XUnitConsoleRunner |

Ideally we will switch to VSTest for netcoreapp and netfx in the future.

cc @RussKeldorph @cshung @danmosemsft

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -28,6 +29,7 @@
     <MicrosoftDotNetGenAPIPackageVersion>1.0.0-beta.19202.3</MicrosoftDotNetGenAPIPackageVersion>
     <MicrosoftDotNetGenFacadesPackageVersion>1.0.0-beta.19177.11</MicrosoftDotNetGenFacadesPackageVersion>
     <MicrosoftDotNetXUnitExtensionsPackageVersion>2.4.0-beta.19202.3</MicrosoftDotNetXUnitExtensionsPackageVersion>
+    <MicrosoftDotNetXUnitConsoleRunnerPackageVersion>2.4.0-beta.19202.3</MicrosoftDotNetXUnitConsoleRunnerPackageVersion>
```
Comment (hoyosjs) [inline eng/Versions.props]: Is this the right version? I see 2.5.1-beta.19202.3 as the version corresponding to this arcade hash.

Comment (ViktorHofer) [reply eng/Versions.props]: This is just a placeholder and will be auto-updated by maestro.

Comment (ViktorHofer) [reply eng/Versions.props]: I just updated the version with a manual darc update. Thanks!

**Comment by ViktorHofer** [comment]
Errors are unrelated: https://github.com/dotnet/corefx/issues/36594

**Pull request merged**